drivers/watchdog/wdt_gecko.c

```diff
@@ -259,6 +259,8 @@ static int wdt_gecko_init(const struct device *dev)
 	CMU_ClockEnable(config->clock, true);
 #else
 	CMU_ClockSelectSet(config->clock, cmuSelect_ULFRCO);
+	/* Enable Watchdog clock. */
+	CMU_ClockEnable(config->clock, true);
```

Comment:
One more simplification. As the clock is now enabled in both cases, we can take it out of the `#ifdef`s.

```diff
- #if !defined(_SILICON_LABS_32B_SERIES_2)
- 	/* Ensure LE modules are clocked */
- 	CMU_ClockEnable(config->clock, true);
- #else
+ #ifdef _SILICON_LABS_32B_SERIES_2
	CMU_ClockSelectSet(config->clock, cmuSelect_ULFRCO);
+ #endif
+ 
	/* Enable Watchdog clock. */
	CMU_ClockEnable(config->clock, true);
- #endif
```

I don't know this MCU at all. Is it correct that the `CMU_ClockSelectSet` call is only necessary for a particular series? The `ULFRCO` is enabled for all cases, so this looks a bit strange. Can you please double-check?

Reply:
Simplified the code as requested